opus/application/static_media/js/browse.js

```diff
@@ -675,56 +674,86 @@ var o_browse = {
+            // (account for height of thead)
             topBoxBoundary = (topBoxBoundary ||
-                             (opus.prefs[browse]  === "gallery") ?
-                                $(`${tab} .gallery-contents`).offset().top :
-                                $(`${tab} .gallery-contents`).offset().top + $(`${tab} .op-data-table thead th`).outerHeight());
-
-            // compare the image .top + half its height in order to make sure we account for partial images
-            let topBox = $(elem).offset().top + $(elem).height()/2;
-            if (topBox >= topBoxBoundary) {
-                let obsNum = $(elem).data("obs");
-
-                // update obsNum in both infiniteScroll instances
-                // store the most top left obsNum in gallery for both infiniteScroll instances, this will be used to updated slider obsNum
-                $(`${tab} .op-gallery-view`).infiniteScroll({"obsNum": obsNum});
-                if (contentsView === ".op-data-table-view") {
-                    obsNum = Math.floor((obsNum-1)/o_browse.gallerySliderStep+0.0000001)*o_browse.gallerySliderStep+1;
+                              (opus.prefs[browse]  === "gallery") ?
+                              $(`${tab} .gallery-contents`).offset().top :
+                              $(`${tab} .gallery-contents`).offset().top + $(`${tab} .op-data-table thead th`).outerHeight());
+
+            // table: obsNum = calculatedFirstObs + number of row
+            // gallery: obsNum = calculatedFirstObs + number of row * number of obs in a row
+            let obsNumDiff = ((opus.prefs[browse] === "dataTable") ?
+                              Math.round((topBoxBoundary - firstCachedObsTop)/$(`${tab} tbody tr`).outerHeight()) :
+                              Math.round((topBoxBoundary - firstCachedObsTop)/o_browse.imageSize) * galleryBoundingRect.x);
+            let obsNum = obsNumDiff + calculatedFirstObs;
+
+            let numToDelete = ((galleryBoundingRect.x - (firstCachedObs - 1) % galleryBoundingRect.x) %
+                               galleryBoundingRect.x);
+
+            let galleryObsElem = $(`${tab} .gallery [data-obs]`);
+            let tableObsElem = $(`${tab} .op-data-table-view [data-obs]`);
+            // delete first "numToDelete" obs if row size is changed
+            if (numToDelete !== 0) {
+                for (let count = 0; count < numToDelete; count++) {
+                    o_browse.deleteCachedObservation(galleryObsElem, tableObsElem, count);
                 }
+            }
 
-                $(`${tab} .op-data-table-view`).infiniteScroll({"obsNum": obsNum});
-                opus.prefs[startObsLabel] = obsNum;
+            // Update obsNum in both infiniteScroll instances.
+            // Store the most top left obsNum in gallery for both infiniteScroll instances
+            // (this will be used to updated slider obsNum).
+            if (contentsView === ".op-data-table-view") {
```

Comment:
i find it confusing to check the container class vs. whether this is gallery or table view.

Reply:
Agreed